core/src/main/scala/chisel3/experimental/dataview/package.scala

```diff
@@ -103,10 +103,21 @@ package object dataview {
       val tex = unfoldView(te).find(targetContains).getOrElse(err("Target", te))
       val vex = unfoldView(ve).find(viewFieldLookup.contains).getOrElse(err("View", ve))
 
-      if (tex.getClass != vex.getClass) {
-        val fieldName = viewFieldName(vex)
-        throw InvalidViewException(s"Field $fieldName specified as view of non-type-equivalent value $tex")
+      (tex, vex) match {
+        /* Allow views where the types are equal. */
+        case (a, b) if a.getClass == b.getClass =>
+        /* Allow UInt <=> Reset views. */
+        case (a: UInt, _: Reset) if a.isWidthKnown && a.getWidth == 1 => true
+        case (_: Reset, a: UInt) if a.isWidthKnown && a.getWidth == 1 => true
+        /* Allow AsyncReset <=> Reset views. */
+        case (a: AsyncReset, _: Reset) => true
+        case (_: Reset, a: AsyncReset) => true
+        /* All other views produce a runtime error. */
+        case _ =>
```

Comment:
The return values are unnecessary, also should this be UInts with checked widths or just Bool? Generally we say that synchronous reset is Bool, not 1-bit UInt.

Reply:
I didn't realize that it was restricted to `Bool` only as opposed to `UInt(1.W)`. Good catch. I'll change this.